KEEP/bin/butch-purge

```diff
@@ -0,0 +1,55 @@
+#!/bin/sh
+#set -x
+if [ -z "$S" ] ; then
+	echo 'S not set, did you source config?'
+	exit 1
+fi
+
+usage() {
+cat << EOF
+butch purge : remove packages and their now orphaned dependencies
+- will remove the supplied packages
+- dependencies of these packages will also be removed if
+  nothing installed depends on them
+- you will be prompted with an interactive confirmation
+  of the files to remove prior to their execution
+
+these operations require root access
+use with care, it will remove packages!
+
+synopsis:
+butch prep --target1 [--target2 ... --targetN]
+EOF
+exit 1
+}
+
+[ -z "$1" ] && usage
+[ "$1" = --help ] && usage
+
+tmp=$(mktemp -u)
+for pkg in $(butch purgelist "$@") ; do
+	echo $pkg >> $tmp
+done
+cat <<- EOF >> $tmp
+#
+# welcome to butch's interactive purge mode.
+#
+# all lines represent a package to be removed.
+#
+# remove any packages you wish to keep and then save.
+#
+EOF
+
+[ -z "$EDITOR" ] && EDITOR=vi
+$EDITOR $tmp
+
+trash=""
+while read i ; do
+	printf "%s\n" "$i" | grep -v '^#' >/dev/null 2>&1 && \
+	trash="$trash $i"
+done < $tmp
+rm -f $tmp
+
+[ -z "$trash" ] && exit 0
+
+butch rm $trash
```

Comment:
i commented this already on your prev. commits but here goes again: we need to call butch binaries via BUTCH_BINDIR like the other packages do. rationale in my prev comment.